plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/OpenLineageListener.java

```diff
@@ -278,17 +283,24 @@ private RunFacetsBuilder getBaseRunFacetsBuilder(QueryContext queryContext)
                         .build());
     }
 
-    private JobBuilder getBaseJobBuilder(QueryMetadata queryMetadata)
+    private JobBuilder getBaseJobBuilder(QueryContext queryContext, QueryMetadata queryMetadata)
     {
         return openLineage.newJobBuilder()
                 .namespace(this.jobNamespace)
-                .name(queryMetadata.getQueryId())
+                .name(buildJobName(queryContext, queryMetadata))
                 .facets(openLineage.newJobFacetsBuilder()
                             .jobType(openLineage.newJobTypeJobFacet("BATCH", "TRINO", "QUERY"))
                             .sql(openLineage.newSQLJobFacet(queryMetadata.getQuery(), "trino"))
                             .build());
     }
 
+    private String buildJobName(QueryContext queryContext, QueryMetadata queryMetadata)
+    {
+        FormatInterpolator<OpenLineageJobContext> interpolator = new FormatInterpolator<>(jobNameFormat, OpenLineageJobInterpolatedValues.values());
```

Comment:
This can be a extracted as a final variable right ?

Reply:
Yes, done